src/sentry/search/events/builder.py

```diff
@@ -114,18 +114,43 @@ def validate_having_clause(self) -> None:
         Skipped if auto_aggregations are enabled, and at least one other aggregate is selected
         This is so we don't change grouping suddenly
         """
+        # construct the list of each condition, this is so many steps cause self.having can be a mix of
+        # BooleanConditions and Conditions.
+        to_check: List[Condition] = []
+        boolean_conditions: List[BooleanCondition] = []
+        for condition in self.having:
+            if isinstance(condition, Condition):
+                to_check.append(condition)
+            elif isinstance(condition, BooleanCondition):
+                boolean_conditions.append(condition)
+
+        while len(boolean_conditions) > 0:
+            boolean_condition = boolean_conditions.pop()
+            for condition in boolean_condition.conditions:
+                if isinstance(condition, Condition):
+                    to_check.append(condition)
+                elif isinstance(condition, BooleanCondition):
+                    boolean_conditions.append(condition)
+
         if self.auto_aggregations and self.aggregates:
+            for condition in to_check:
+                lhs = condition.lhs
+                if isinstance(lhs, CurriedFunction) and lhs not in self.columns:
+                    self.columns.append(lhs)
+                    self.aggregates.append(lhs)
```

Comment:
Why do `lhs` need to be added to `self.aggregates`?

Reply:
Not used anywhere, added this for consistency since `lhs` is an aggregation